src/productcatalogservice/server.go

```diff
@@ -148,6 +168,22 @@ type productCatalog struct{}
 func parseCatalog() []*pb.Product {
 	var cat pb.ListProductsResponse
 
+	if problem {
+		done := make(chan int)
```

Comment:
why not just `time.Sleep()` here? Are you trying to trigger a 100% memory?

If so, please simplify as 

	t := time.Now().Add(150 * time.Millisecond)
	for time.Now().Before(t) {}

Reply:
Yes, that's what I'm trying to do. Simplified as suggested.